### PR TITLE
Follow logs automatically

### LINKED ISSUE #1061: Scrollbar visible when log doesn't fill screen
screen shot 2016-12-21 at 10 37 52 am

### PR DESCRIPTION
**Pull request by spadgett**
* Follow logs as soon as the tab is selected.
* Always show follow log links for live logs. This lets you follow or
  stop following without having to wait for the log to go past the
  bottom of the visible page. (The go to top and go to end links will
  still be hidden unless the log is taller than the visible page.)
* Fix issue with affix-bottom being set on the follow log link, causing
  the link to sometimes appear at the bottom of the page.
* Fix references to unused `$scope.autoScroll` variable. It has since
  been renamed `$scope.autoScrollActive`.
* Fix issue where scrollbars appear even if the log doesn't fill the page.

Fixes #1061

@stevekuznetsov fyi

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Cool!

File: app/scripts/directives/logViewer.js [hunk 1]
```diff
@@ -197,7 +194,8 @@ angular.module('openshiftConsole')
               var fill = $scope.fixedHeight ? $scope.fixedHeight : Math.floor($(window).height() - contentTop - pulserHeight);
               if (!$scope.chromeless && !$scope.fixedHeight) {
                 // Add some bottom margin if not chromeless.
```
Comment (benjaminapetersen) [inline app/scripts/directives/logViewer.js]: I cant think of a clever way to do it, but I wish we could share variable values between JS & LESS. Hmm.

File: app/styles/_variables.less [hunk 2]
```diff
@@ -40,6 +40,8 @@
 @link-hover-color-bright: lighten(@link-color, 13%);
 @list-view-expanded-bg: #f5f5f5;
 @log-bg-color: #101214;
+// The log viewer needs to be udpated if this value changes.  It accounts for
```
Comment (benjaminapetersen) [inline app/styles/_variables.less]: Glad you added this comment!

File: app/views/directives/logs/_log-viewer.html [hunk 3]
```diff
@@ -68,14 +68,13 @@
 <div ng-show="state=='logs'">
   <div class="log-view" id="{{logViewerID}}" ng-class="{'log-fixed-height': fixedHeight}">
```
Comment (benjaminapetersen) [inline app/views/directives/logs/_log-viewer.html]: Just noticed this, for dynamic id on line 69, are we supposed to use `ng-attr-id`?  Its not part of your PR (and I bet I overlooked this initially) so prob not a big deal.

File: app/scripts/directives/logViewer.js (hunk 1 above)
Comment (spadgett) [inline app/scripts/directives/logViewer.js]: Yeah, I can't find a cleaner way. I was thinking of trying to get the padding/margin on the element in JS, but the element with the margin keeps changing.

**Comment by spadgett** [comment]
[merge]

**Pull request merged**